plugins/speccy.py

```diff
@@ -0,0 +1,123 @@
+
+GPU_RE = re.compile('.*(amd|radeon|intel|integrated|nvidia|geforce|gtx).*\n.*', re.I)
+PICO_RE = re.compile('.*pico', re.I)
+KMS_RE = re.compile('.*kms', re.I)
+BOOSTER_RE = re.compile('.*booster', re.I)
+REVIVER_RE = re.compile('.*reviver', re.I)
+KILLER_RE = re.compile('.*killer.+service', re.I)
+
+
+@hook.regex(url_re)
+def get_speccy_url(match):
+
+    with closing(requests.get(match.group(), )) as response:
+        soup = BS(response.content, "lxml", from_encoding=response.encoding)
+
+    body = soup.body
+
+    data = []
+
+    def specout():
+        spec_out = re.sub(r"\s+", " ", ' ● '.join(data))
+        return spec_out
+
+    os_spec = body.find("div", text='Operating System')
+    if os_spec:
+        data.append(
+            "\x02OS:\x02" + " " + os_spec.next_sibling.next_sibling.text)
+
+    def cspace():
+        c_drive = body.find("div", class_="datavalue", text='C:')
+        if c_drive is not None:
+            c_drive = body.parent.parent
+            c_free_space_data = c_drive.find(
+                "div", text='Free Space:\xA0', class_="datakey").parent
+            return c_free_space_data.find(class_="datavalue").text
+
+    if cspace():
+        data.append("\x02Space Left (C:\):\x02" + " " + cspace())
+
+    ram_spec = body.find("div", text='RAM')
+    if ram_spec:
+        data.append(
+            "\x02RAM:\x02" + " " + ram_spec.next_sibling.next_sibling.text)
+
+    ram_usg_spec = body.find(
+        "div", class_="blue clear",
+        text='Physical Memory').next_sibling.next_sibling.find(
+            "div", text='Memory Usage:\xA0',
+            class_="datakey").parent.find(class_="datavalue").text
+    if ram_usg_spec:
+        data.append("\x02RAM Usg:\x02" + " " + ram_usg_spec)
+
+    cpu_spec = body.find("div", text='CPU')
+    if cpu_spec:
+        data.append(
+            "\x02CPU:\x02" + " " + cpu_spec.next_sibling.next_sibling.text)
+
+    gpu_find = body.find("div", text='Graphics').next_sibling.next_sibling.text
+    gpu_spec = ""
+    for gpu_string in GPU_RE.finditer(gpu_find):
+        gpu_spec += gpu_string.group()
+    if gpu_spec:
+        data.append("\x02GPU:\x02" + " " + gpu_spec)
+
+    pico_spec = body.find("div", text=PICO_RE)
+    if pico_spec:
+        data.append("\x02Badware:\x02" + " " + pico_spec.text)
+
+    kms_spec = body.find("div", text=KMS_RE)
+    if kms_spec:
+        data.append("\x02Badware:\x02" + " " + kms_spec.text)
+
+    booster_spec = body.find("div", text=BOOSTER_RE)
+    if booster_spec:
+        data.append("\x02Badware:\x02" + " " + booster_spec.text)
+
+    reviver_spec = body.find("div", text=REVIVER_RE)
+    if reviver_spec:
+        data.append("\x02Badware:\x02" + " " + reviver_spec.text)
+
+    killer_spec = body.find("div", text=KILLER_RE)
+    if killer_spec:
+        data.append("\x02Badware:\x02" + " " + killer_spec.text)
+
+    if 'Badware' not in specout():
+        data.append("\x02No Badware\x02")
```

Comment:
Might be easier to just set a variable like `has_badware = False` and set that to `True` whenever you find something that's considered badware.